tidy3d/components/parameter_perturbation.py

```diff
@@ -1227,6 +1228,389 @@ def _sample_delta_eps_delta_sigma(
         return delta_eps_sampled, delta_sigma_sampled
 
 
+class AbstractDeltaModel(Tidy3dBaseModel):
+    """Abstract class for the definition of delta models"""
+
+    @abstractmethod
+    def delta_k(self) -> ChargePerturbationType:
+        """Return the perturbation range of the model."""
+
+    @abstractmethod
+    def delta_n(self) -> HeatPerturbationType:
+        """Return the perturbation range of the model."""
+
+
+class NedeljkovicSorefMashanovich(AbstractDeltaModel):
+    """Nedeljkovic-Soref-Mashanovich model for the perturbation of the refractive index and
+    extinction coefficient due to free carriers.
+
+    M. Nedeljkovic, R. Soref and G. Z. Mashanovich, "Free-Carrier Electrorefraction and Electroabsorption
+    Modulation Predictions for Silicon Over the 1–14- μm Infrared Wavelength Range," in IEEE Photonics
+    Journal, vol. 3, no. 6, pp. 1171-1180, Dec. 2011, doi: 10.1109/JPHOT.2011.2171930
+
+    Example
+    -------
+    """
+
+    perturb_coeffs = xr.Dataset(
```

Comment:
I wonder if we are better off with not having `perturb_coeffs` field in this class and store outside of class as, say, `NSM_DATA`